common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		updateButtonVisibility();
+	}
+
+	@Override
+	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
+		if (state == menu)
+			for (MenuItemsDivision div : divisionsList) {
+				for (MenuItem item : div.getList()) {
+					if (item.mousePressed()) {
+						item.playPressSound(this.mc.getSoundHandler());
+						this.pressedItem(item);
+						updateButtonVisibility();
+						return;
+					}
+				}
+			}
+		for (GuiGuideBookTabButton tab : tabList) {
+			if (tab.mousePressed(mc, mouseX, mouseY)) {
+				tab.playPressSound(this.mc.getSoundHandler());
+				if (!tab.isSet && state != menu) {
+					PageInformation setPage = new PageInformation();
+					setPage.page = currentPage.page;
+					setPage.progress = currentPage.progress;
+					if (savedTabs.addTab(new SavedTab(setPage, state))) {
+						tab.isSet = true;
+						if (tab.isSet) tab.isActive = true;
+					}else{
+						return;
+					}
+				} else if(tab.isSet && !tab.isActive){
+					SavedTab info = new SavedTab(savedTabs.tabs.get(tab.id).page, savedTabs.tabs.get(tab.id).state);
+					state = page;
+					currentPage.set(info);
+				}
+				updateButtonVisibility();
+				return;
+			}
+		}
+		super.mouseClicked(mouseX, mouseY, mouseButton);
+		updateButtonVisibility();
+	}
+
+	private void pressedItem(MenuItem item) {
+		selectChapter(item);
+	}
+
+	protected void selectChapter(MenuItem item) {
+		currentPage.set(item);
+		state = page;
+		updateTitle();
+		updateButtonVisibility();
+	}
+
+	protected void updateTitle() {
+		String title = "";
+		title += gbc.getTitle();
+		if (state == menu) title += ": Menu";
+		else {
+			title += ": " + gbc.getDivision(currentPage.getDivision()).getTitle();
+			if (currentPage.getChapter() != -1) {
+				title += " - " + gbc.getDivision(currentPage.getDivision()).getChapter(currentPage.getChapter()).getTitle();
+			}
+		}
+		this.title = title;
+	}
+
+	protected void updateButtonVisibility() {
+		this.prevPage.enabled = currentPage.getIndex() != 0;
+		this.prevPage.visible = state == page;
+		this.nextPage.enabled = currentPage.getIndex() < currentPage.getPageCount() - 1;
+		this.nextPage.visible = state == page;
+		this.home.visible = state != menu;
+		for(GuiGuideBookTabButton tab: tabList){
+			tab.visible = false;
+			tab.isActive = false;
+		}
+		for (int i = 0; i < savedTabs.tabs.size(); i++) {
+			tabList.get(i).visible = true;
+			tabList.get(i).isSet = true;
+			tabList.get(i).isActive = equals(savedTabs.tabs.get(i), new SavedTab(currentPage, state));
+		}
+		if (savedTabs.tabs.size() < maxTabs) tabList.get(savedTabs.tabs.size()).visible = true;
+	}
+
+	protected void nextPage() {
+		if (nextPage.enabled) currentPage.nextPage();
+		updateButtonVisibility();
+	}
+
+	protected void prevPage() {
+		if (prevPage.enabled) currentPage.prevPage();
+		updateButtonVisibility();
+	}
+
+	/* *********************************************** draw functions with a twist *********************************************** */
+
+	/*
```

Comment:
If you have a doc on a method, please use JavaDoc notation then: `/** ... */`

Reply:
Hummm so can I actually use JavaDoc notation on these ones? I just avoided it altogether

Reply:
If you have documentation for a specific method/class/field, then yes, use JavaDoc comments. They won't work for comments that are supposed to target a section of methods, like the "draw functions" comment here, since they will only apply to the method directly following the comment.

Reply:
Makes sense, and tbh I only used to use JavaDoc everywhere because it was green text and normal comments were very hard to see, I have changed IDEA since but yeah :/